### PR TITLE
OpenSearch Plugin: Change HttpClient::sendAsync to synchronous HttpClient::send to get around the java security policy exception

### PR DESCRIPTION
**Pull request by santhoshsram**
## What does this PR do?
The PR changes the async http call made to Infino server to a sync call. The async call was spawning off new threads which were being blocked by opensearch's strict security policy that forbids thread operations. Changing the call to a synchronous send does not trigger the security policy exception.

Changing the async call to sync is ok since each sync call is still made in a separate thread. So the plugin itself will not block on send() call.

### Curious case of threads in OpenSearch
The thread that we manually spawn off in the plugin in InfinoRestHandler (when processing incoming requests) seems to work fine. However the thread that HttpClient::sendAsync() spawns internally runs into OpenSearch's security manager rules. My hypotheses here is that OpenSearch's security manager grants access to modify only the current thread group or the current thread group's parent - check [this comment in OpenSearch source code](https://github.com/opensearch-project/OpenSearch/blob/76ae14a4f2e99592610d3181543f5036214ceb7a/libs/secure-sm/src/main/java/org/opensearch/secure_sm/SecureSM.java#L60) and the sendAsync() call is perhaps running in a different thread group context and is trying to access/modify OpenSearch's threadgroup. Still not fully clear and will need further investigation. **But for now since we have a tested workaround, we'll go ahead with this to make progress.**

## Checklist
- [x] I have built and tested the changes end to end on infino server ↔ opensearch + infino plugin
- [x]  I have written the necessary comments.
- [x]  I have updated the necessary unit tests and integration tests. (for non-documentation changes)

### CODE REVIEW

**Review by vinaykakade — APPROVED** [review]
Thanks for the detailed investigation and getting to the root cause/fix!

File: plugins/infino-opensearch-plugin/src/main/java/org/opensearch/infino/InfinoRestHandler.java [hunk 1]
```diff
@@ -356,15 +350,24 @@ private void sendRequestWithBackoff(HttpClient backoffHttpClient, HttpRequest re
...
-                        attempt, request))
-                .exceptionally(e -> handleException(e, channel, client, indexName, method));
+        try {
+            HttpResponse<String> response = backoffHttpClient.send(request, HttpResponse.BodyHandlers.ofString());
+            processResponse(backoffHttpClient, response, channel, client, indexName, method, attempt, request);
+        } catch (Exception e) {
+            handleException(e, channel, client, indexName, method);
+        }
+
+        // CompletableFuture<Void> future = backoffHttpClient.sendAsync(request,
```
Comment (ekechinwokah) [inline plugins/infino-opensearch-plugin/src/main/java/org/opensearch/infino/InfinoRestHandler.java:360]: Remove this commented-out code.

File: plugins/infino-opensearch-plugin/src/main/plugin-metadata/plugin-security.policy [hunk 2]
```diff
@@ -1,5 +1,5 @@
 grant {
     // Add permissions to allow Infino OpenSearcg Plugin to access Infino using OpenSearch threadpool
     permission org.opensearch.secure_sm.ThreadPermission "modifyArbitraryThread";
-    permission java.net.URLPermission "http://localhost:3000/-", "*";
+    permission java.net.URLPermission "http://*:3000/-", "*";
```
Comment (ekechinwokah) [inline plugins/infino-opensearch-plugin/src/main/plugin-metadata/plugin-security.policy:4]: While we are in this code, should we make the port number configurable? e.g. via an environment variable? Also happy to punt on this for now.

**Review by ekechinwokah — COMMENTED** [review]
Thanks for working on this. Good stuff.

File: plugins/infino-opensearch-plugin/src/main/plugin-metadata/plugin-security.policy (hunk 2 above)
Comment (santhoshsram) [reply plugins/infino-opensearch-plugin/src/main/plugin-metadata/plugin-security.policy]: I updated the policy to allow "http://*:*/-", i.e. access to _**any host**_ on _**any port**_. IMO it is more flexible to leave the policy at the JVM level to be least restrictive and apply any restrictions at the container networking level.

File: plugins/infino-opensearch-plugin/src/main/java/org/opensearch/infino/InfinoRestHandler.java (hunk 1 above)
Comment (santhoshsram) [reply plugins/infino-opensearch-plugin/src/main/java/org/opensearch/infino/InfinoRestHandler.java]: Done. Thanks for catching this.

**Pull request merged**